contracts/interstake-yield-generator/src/migration.rs

```diff
@@ -0,0 +1,42 @@
+use schemars::JsonSchema;
+use semver::Version;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::{Addr, Decimal, DepsMut, Timestamp};
+use cw_storage_plus::Item;
+
+use crate::error::ContractError;
+use crate::msg::MigrateMsg;
+use crate::state::{Config, CONFIG};
+
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct ConfigV0_1_5 {
+    pub owner: Addr,
+    // pub staking_addr: String, // this field is removed in 0.1.5
+    pub team_commission: Decimal,
+    pub denom: String,
+}
+
+pub fn migrate_config(
+    deps: DepsMut,
+    version: &Version,
+    msg: MigrateMsg,
+) -> Result<(), ContractError> {
+    if *version < "0.1.4".parse::<Version>().unwrap() {
+        let old_storage: Item<ConfigV0_1_5> = Item::new("config");
+        let config = old_storage.load(deps.storage)?;
+
+        let treasury = deps.api.addr_validate(msg.treasury.as_str())?;
+        let new_config = Config {
+            owner: config.owner,
+            treasury,
+            restake_commission: config.team_commission,
+            transfer_commission: msg.transfer_commission,
+            denom: config.denom,
+            unbonding_period: Timestamp::from_seconds(3600 * 24 * 28), // default 28 days
```

Comment:
Nit: It's not a case here, but worth pointing out - it's better to take that value from actual config in case we'd be already on multiple chains.

Reply:
What do you mean with the actual config?

Reply:
You're hardcoding those 28 days here instead of taking them from `config.unbonding_period`.

Reply:
Oh wow, i actually missed that 🤦